src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,412 @@
+        dialog.panel.valueField1.setText(getCredential(PROPERTY_API_KEY));
+        dialog.panel.valueLabel2.setVisible(false);
+        dialog.panel.valueField2.setVisible(false);
+        dialog.panel.temporaryCheckBox.setSelected(
+                isCredentialStoredTemporarily(PROPERTY_API_KEY));
+        dialog.show();
+    }
+
+    @Override
+    protected String translate(Language sLang, Language tLang, String text)
+            throws Exception {
+
+        String mthubKey = getCredential(PROPERTY_API_KEY);
+        if (mthubKey == null || mthubKey.isEmpty()) {
+            return OStrings.getString("MT_ENGINE_MTHUB_KEY_NOTFOUND");
+        }
+
+        String prev = getFromCache(sLang, tLang, text);
+        if (prev != null) {
+            return prev;
+        }
+
+        if (this.availableLanguageCodes == null) {
+            this.getAvailableLanguageCodes();
+            if (this.availableLanguageCodes == null) {
+                /*error to obtain the available language,
+                it is a error with the user key*/
+                return OStrings.getString("MT_ENGINE_MTHUB_INVALID_KEY");
+            }
+        }
+
+        String encoding = StandardCharsets.UTF_8.name();
+        try {
+            URLConnection connection = new URL(GT_URL).openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestProperty("Accept-Charset", encoding);
+            connection.setRequestProperty("Content-Type",
+                                        "application/json;charset=" + encoding);
+
+            try (OutputStream output = connection.getOutputStream()) {
+
+                OM.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY,
+                        true);
+                List<String> segments = Collections.singletonList(text.substring(0,
+                                 Math.min(text.length(), LIMIT_CHARACTER - 1)));
+
+                MTHUBTranslateRequest mttr = new MTHUBTranslateRequest(
+                        mthubKey, normaliseCode(sLang), normaliseCode(tLang),
+                        segments);
+
+                String json = OM.writeValueAsString(mttr);
+                output.write(json.getBytes(encoding));
+
+                MTHUBTranslateResponse response = OM.readValue(
+                                                  connection.getInputStream(),
+                                                  MTHUBTranslateResponse.class);
+
+                String translation = "";
+
+                if (response.isSuccess()) {
+                    translation = response.getData().
+                                get(0).getSegments().get(0).get(0).
+                                getTranslation();
+                }
+
+                putToCache(sLang, tLang, text, translation);
+                return translation;
+            } catch (Exception e) {
+                if (e.getMessage().contains("401")) {
```

Comment:
I must insist that parsing the exception message for error codes is not acceptable. Please get the error code directly from the `connection` object per my [earlier comment](https://github.com/omegat-org/omegat/pull/31#discussion_r252269454).